src/test/scala/group/scala/karazin/circe/literal/extras/arbitraries/ArbitrariesJavaDateTime.scala

```diff
@@ -0,0 +1,81 @@
+package group.scala.karazin.circe.literal.extras
+
+import cats.implicits._
+import io.circe.syntax._
+import io.circe.{Json, JsonNumber, JsonObject, parser}
+import org.scalacheck.Gen
+import org.scalacheck.Arbitrary
+import org.scalacheck.Prop._
+import io.circe.Encoder.encodeJsonObject
+import java.time._
+import scala.jdk.CollectionConverters._
+
+
+trait ArbitrariesJavaDateTime {
+  val instantGenerator: Gen[Instant] = Gen.choose(Instant.MIN, Instant.MAX)
+
+  val periodGenerator: Gen[Period] =
+    for {
+      day <- Gen.posNum[Int]
+    } yield Period.ofDays(day)
+
+  val zoneIdGenerator: Gen[ZoneId] = {
+    for {
+      zone <- Gen.oneOf((ZoneId.getAvailableZoneIds).asScala)
+    } yield ZoneId.of(zone)
+
+  }
+
+  val localDateGenerator: Gen[LocalDate] = Gen.choose(LocalDate.MIN, LocalDate.MAX)
+
+  val localTimeGenerator: Gen[LocalTime] = Gen.choose(LocalTime.MIN, LocalTime.MAX)
+
+  val localDateTimeGenerator: Gen[LocalDateTime] = Gen.choose(LocalDateTime.MIN, LocalDateTime.MAX)
+
+  val monthDayGenerator: Gen[MonthDay] =
+    for {
+      month <- Gen.choose(0, 11)
+      day   <- Gen.choose(1, LocalDate.of(2020, month, 1).getDayOfMonth)
+    } yield MonthDay.of(month, day)
+
+  val offsetTimeGenerator: Gen[OffsetTime] = Gen.choose(OffsetTime.MIN, OffsetTime.MAX)
+
+  val offsetDateTimeGenerator: Gen[OffsetDateTime] = Gen.choose(OffsetDateTime.MIN, OffsetDateTime.MAX)
+
+  val yearGenerator: Gen[Year] =
+    for {
+      year <- Gen.choose(Year.MIN_VALUE, Year.MAX_VALUE)
+    } yield Year.of(year)
+
+  val yearMonthGenerator: Gen[YearMonth] =
+    for {
+      year  <- Gen.choose(Year.MIN_VALUE, Year.MAX_VALUE)
```

Comment:
Let's reuse `yearGenerator`.

Reply:
i dont really know if it is necessary to do an extra variable for it like because yearGenerator generates `java.time.Year` but not `int` that is needed  here for yearMonthGenerator. Or i understand not correctly